src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -2055,10 +2085,11 @@ void Generator::GenerateClassConstructor(const GeneratorOptions& options,
       " * valid.\n"
       " * @extends {jspb.Message}\n"
       " * @constructor\n"
+      " * $final$\n"
       " */\n"
       "$classprefix$$classname$ = function(opt_data) {\n",
-      "classprefix", GetMessagePathPrefix(options, desc), "classname",
-      desc->name());
+      "final", final, "classprefix", GetMessagePathPrefix(options, desc),
```

Comment:
I think this is a breaking change, it will make all apps jspb message final. I'm surprised to see the generator changes appearing in the open source, is this because the sync was started after cl/303150636 was submitted?

Reply:
Fixed.